### PR TITLE
Use js-ilp and the new SPSP flow

### PR DESCRIPTION
**Pull request by vhpoet**
This is a work in progress, opening a PR to start discussions.

### CODE REVIEW

**Comment by vhpoet** [comment]
@emschwartz Here's how I think about the SPSP flow in the wallet. Correct me if I'm doing something wrong.

1. `wallet1:` user enters the destination address `alice@wallet2.com`
2. `wallet1:` webfinger lookup gets the `receiverUri`
3. `wallet1:` GET `receiverUri`: gets receiver info from `wallet2` (query)
4. `wallet2:` respond with a receiverinfo
5. `wallet1:` user enters the `source` or `destination` amount
6. `wallet1:` THIS STEP IS ONLY FOR THE SOURCE AMOUNT. If the amount is `source`, call `sender.quoteSourceAmount` and then assign the result to `destinationAmount`
7. `wallet1:` POST `receiverUri`: with the `destinationAmount` (setup)
8. `wallet2:` call `receiver.createRequest` with the `destinationAmount` and return the result
9. `wallet1:` user sees the quote result, decides that she wants to go with it, adds a message and clicks send (user could be changing the amount multiple times and then lastly adding a message, so there could be multiple step 7s)
10. `wallet1:` `sender.payRequest`
11. `wallet2:` the listener from step 7 catches the payment fulfills it and changes the state in the db

**Comment by vhpoet** [comment]
don't feel great about using the `setup` step as a `requestQuote` step and calling it multiple times. Talked to Stefan about this and it seems like we have two options for a clean solution: Take out quoting from the `js-ilp` completely (including `sender.quoteSourceAmount`) and use `js-ilp-core` for quoting, or implement `sender.quoteDestinationAmount` on `js-ilp`.

File: api/src/controllers/payments.js [hunk 1]
```diff
@@ -261,37 +269,53 @@ function PaymentsControllerFactory (Auth, Payment, log, ledger, config, utils) {
...
       // Create the payment object
-      let payment = new Payment()
+      const payment = new Payment()
       payment.setDataExternal(paymentObj)
 
       try {
         yield payment.create()
-      } catch(e) {
+      } catch (e) {
+        console.log('payments:299', 'woops', e)
```
Comment (emschwartz) [inline api/src/controllers/payments.js]: ?

**Comment by emschwartz** [comment]
I like the idea of adding `sender.quoteDestinationAmount`. See https://github.com/interledger/js-ilp/pull/33

File: api/src/controllers/payments.js (hunk 1 above)
Comment (vhpoet) [inline api/src/controllers/payments.js]: @emschwartz don't worry about the code too much yet, it's not a final commit.

**Pull request merged**